google/cloud/aiplatform/datasets/tabular_dataset.py

```diff
@@ -33,6 +39,151 @@ class TabularDataset(datasets._Dataset):
         schema.dataset.metadata.tabular,
     )
 
+    @property
+    def column_names(self) -> List[str]:
+        """Retrieve the columns for the dataset by extracting it from the Google Cloud Storage or 
+        Google BigQuery source.
+
+        Returns:
+            List[str]
+                A list of columns names
+                
+        Raises:
+            RuntimeError: When no valid source is found.
+        """
+
+        metadata = self._gca_resource.metadata
+
+        if metadata is None:
+            raise RuntimeError("No metadata found for dataset")
+
+        input_config = metadata.get("inputConfig")
+
+        if input_config is None:
+            raise RuntimeError("No inputConfig found for dataset")
+
+        gcs_source = input_config.get("gcsSource")
+        bq_source = input_config.get("bigquerySource")
+
+        if gcs_source:
+            gcs_source_uris = gcs_source.get("uri")
+
+            if gcs_source_uris and len(gcs_source_uris) > 0:
+                # Lexicographically sort the files
+                gcs_source_uris.sort()
+
+                # Get the first file in sorted list
+                return TabularDataset._retrieve_gcs_source_columns(
+                    self.project, gcs_source_uris[0]
+                )
+        elif bq_source:
+            bq_table_uri = bq_source.get("uri")
+            if bq_table_uri:
+                return TabularDataset._retrieve_bq_source_columns(
+                    self.project, bq_table_uri
+                )
+
+        raise RuntimeError("No valid CSV or BigQuery datasource found.")
+
+    @classmethod
+    def _retrieve_gcs_source_columns(
```

Comment:
These next two methods seem like they can be static methods.

Reply:
It is generally preferred in Python to use static where possible, then class and then method?

Reply:
In general, though there's probably exceptions, static methods are preferred over class methods when the method doesn't interact with the class. In this case it looks like `cls` is unused. The main reason to choose a static method over a standalone method is to indicate that the method is logically coupled to the class. That seems to make sense in this case because retrieving columns names would is unique to tabular datasets. If we find we need this functionality elsewhere in the SDK then we should implement these methods as standalone method.

Reply:
Sounds good.

Reply:
Done